vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/occupancy_handler.rb

```diff
@@ -16,25 +16,36 @@ def initialize(event)
 
       def process
         if current_occupant? && exiting?
-          existing_occupancy.update!(
-            exit_event: event,
-            exit_at: Time.current,
-          )
+          associate_exit_to(existing_occupancy)
         elsif new_occupant? && entering?
-          new_occupancy.update!(
-            entry_event: event,
-            entry_at: Time.current,
-          )
-        else
-          # TODO: Add error cases
-          Rails.logger.warn "Encountered Double-Scan Situation"
+          associate_entry_to(new_occupancy)
+        elsif new_occupant? && exiting?
+          new_occupancy.mark_orphaned!
+          associate_exit_to(new_occupancy)
+        elsif current_occupant? && entering?
+          existing_occupancy.mark_orphaned!
+          associate_entry_to(new_occupancy)
         end
```

Comment:
There shouldn't be any other potential conditions, but would it be worth adding an `else` that maybe raises an error indicating a bug? Might not be worth it, though.